### PR TITLE
github: bump artifact actions version

### PR DESCRIPTION
**Pull request by agottardo**
Update .github/workflows/build.yml to use v4s of the upload-artifact and download-artifact actions. Adds a workaround for actions/upload-artifact#176.

Co-authored-by: Mario Minardi <mminardi@shaw.ca>

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -60,7 +60,7 @@ jobs:
     - name: archive
       run: cd .. && tar --exclude-vcs -zcf ${{ matrix.GOOS }}-${{ matrix.GOARCH }}.tar.gz go
     - name: save
-      uses: actions/upload-artifact@v1
+      uses: actions/upload-artifact@v4
```
Comment (mpminardi) [inline .github/workflows/build.yml]: If we want to pin:
```suggestion
      uses: actions/upload-artifact@c24449f33cd45d4826c6702db7e49f7cdb9b551d # v4.4.0 
```

Comment (agottardo) [reply .github/workflows/build.yml]: Committed.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -60,7 +60,7 @@ jobs:
     - name: archive
       run: cd .. && tar --exclude-vcs -zcf ${{ matrix.GOOS }}-${{ matrix.GOARCH }}.tar.gz go
     - name: save
-      uses: actions/upload-artifact@v1
+      uses: actions/upload-artifact@c24449f33cd45d4826c6702db7e49f7cdb9b551d # v4.4.0
```
Comment (mpminardi) [inline .github/workflows/build.yml]: Whoops had the wrong sha in my clipboard:
```suggestion
      uses: actions/upload-artifact@50769540e7f4bd5e21e526ee35c689e35e0d6874 # v4.4.0 
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -94,7 +94,7 @@ jobs:
     needs: [create_release]
     steps:
     - name: download artifact
-      uses: actions/download-artifact@v1
+      uses: actions/download-artifact@v4
```
Comment (mpminardi) [inline .github/workflows/build.yml]: If we want to pin this as well:
```suggestion
      uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
```

**Review by mpminardi — APPROVED** [review]

**Pull request merged**